From 482da9131ba56b0291df0bb9d1cdd338643e5f8b Mon Sep 17 00:00:00 2001
From: Robert Edmonds <edmonds@debian.org>
Date: Sun, 9 Aug 2020 20:40:22 -0400
Subject: [PATCH] Enable remote-control by default

---
 util/config_file.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/config_file.c b/util/config_file.c
index 0e9ee47..d2ff94b 100644
--- a/util/config_file.c
+++ b/util/config_file.c
@@ -269,7 +269,7 @@ config_create(void)
 	cfg->insecure_lan_zones = 0;
 	cfg->python_script = NULL;
 	cfg->dynlib_file = NULL;
-	cfg->remote_control_enable = 0;
+	cfg->remote_control_enable = 1; /* Debian specific */
 	cfg->control_ifs.first = NULL;
 	cfg->control_ifs.last = NULL;
 	cfg->control_port = UNBOUND_CONTROL_PORT;
-- 
2.28.0

